File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []

    
    for i in range(int(yellow ** 1/2) + 1, 0, -1):
        if yellow % i == 0 and (i + 2) * (yellow / i + 2) - yellow == brown:
            answer = [i+2, yellow / i + 2]
            break

        
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    for (let i = 1; i <= Math.floor(Math.sqrt(yellow)); i++) {
        if (yellow % i === 0) {
            const col = i + 2
            const row = yellow / i + 2
            
            if (col * row === brown + yellow) answer = [row, col]
        }
    }
    return answer;
}